### PR TITLE
Fix false positive with inference of `http` module

### PR DESCRIPTION
**Pull request by mbyrnepr2**
Fix false positive with inference of `http` module when iterating `HTTPStatus`.

The [Python source code for HTTPStatus](https://github.com/python/cpython/blob/main/Lib/http/__init__.py) inherits from IntEnum which is iterable.

Closes PyCQA/pylint#7307

## Steps

- [ ] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [ ] Write a good description on what the PR does.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |
|    | :sparkles: New feature |
|    | :hammer: Refactoring   |
|    | :scroll: Docs          |

### CODE REVIEW

File: ChangeLog [hunk 1]
```diff
@@ -28,6 +28,10 @@ Release date: TBA
 
   Closes #1737
 
+* Fix false positive with inference of `http` module when iterating `HTTPStatus`.
```
Comment (jacobtylerwalls) [inline ChangeLog:31]: Single backticks seems to just [create italics](https://astroid.pycqa.org/en/latest/changelog.html)
```suggestion
* Fix false positive with inference of ``http`` module when iterating ``HTTPStatus``.
```

**Review by jacobtylerwalls — APPROVED** [review]
Terrific!

File: ChangeLog (hunk 1 above)
Comment (mbyrnepr2) [reply ChangeLog:31]: Thanks!

File: ChangeLog [hunk 2]
```diff
@@ -30,6 +30,9 @@ Release date: TBA
 
   Closes #1737
 
+* Fix false positive with inference of ``http`` module when iterating ``HTTPStatus``.
+
+  Refs PyCQA/pylint#7307
```
Comment (DanielNoord) [inline ChangeLog:35]:
```suggestion
  Refs PyCQA/pylint#7307

```

**Review by DanielNoord — APPROVED** [review]

**Pull request merged**